src/backends/mpv.moon

```diff
@@ -0,0 +1,121 @@
+class MpvBackend extends Backend
+	new: =>
+		@name = "mpv"
+
+	appendProperty: (out, property_name, option_name) =>
+		option_name = option_name or property_name
+		prop = mp.get_property(property_name)
+		if prop and prop != ""
+			append(out, {"--#{option_name}=#{prop}"})
+
+	getPlaybackOptions: =>
+		ret = {}
+		self\appendProperty(ret, "sub-ass-override")
+		self\appendProperty(ret, "sub-ass-force-style")
+		self\appendProperty(ret, "sub-auto")
+
+		-- tracks added manually (eg. drag-and-drop) won't appear on sub-files, so we
+		-- read them from the track-list.
+		for _, track in ipairs mp.get_property_native("track-list")
+			if track["type"] == "sub" and track["external"]
+				append(ret, {"--sub-files-append=#{track['external-filename']}"})
+
+		return ret
+
+	-- Turn `MpvFilter`s into command line options.
+	solveFilters: (filters) =>
+		solved = {}
+		for filter in *filters
+			str = filter.lavfiCompat and "lavfi-" or ""
+			str ..= filter.name .. "="
+			for k, v in pairs filter.params
+				str ..= "#{k}=%#{string.len(v)}%#{v}:"
+			solved[#solved+1] = string.sub(str, 0, string.len(str) - 1)
```

Comment:
The logic for positional parameters (`@0, @1`) isn't implemented here. Not sure if mpv handles it internally (it seems so, would like a link for that just in case).

To be honest, I'd just drop that logic entirely and just use the params.

Reply:
For the record, this is the format mpv returns them in when you get the vf property, so apply_current_filters was actually producing commands with the @0 syntax. In fact, for some reason, the lua table it returns sometimes has the parameters in the wrong order, e.g. `scale=-1:720` was returning parameters like `{ "@1" = "720", " @0" = "-1" }`, the only reason they worked was because mpv seems to understand these.

Something clearer could be done internally though, but then you'd still have to process the @0 syntax when getting the current filters.

Reply:
Oh, ok, didn't know that.